### PR TITLE
PyRETIS: A well-done, medium-sized Python library for rare events

### PR DESCRIPTION
**Pull request by EnricoRiccardi**
Main meta for PyRETIS 1.00

### CODE REVIEW

**Comment by dougalsutherland** [comment]
#3379 and #3378 seem to be the same thing?

**Comment by EnricoRiccardi** [comment]
yes, I did some more tests (that I was thinking I deleted) and then trying to clean up into a final form in #3380. #3379 and #3378 should be removed completely.

**Comment by dougalsutherland** [comment]
You should probably change the name of this PR to something informative, too. :)

File: pyretis/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [py<3]
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+  ## preserve_egg_dir: True
```
Comment (dougalsutherland) [inline pyretis/meta.yaml]: You should probably just delete this comment.

File: pyretis/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  build:
+  - python
+  - setuptools
+  - colorama
+  - tqdm
+  - future
+  - numpy
+  - scipy
+  - matplotlib
+  - jinja2
```
Comment (dougalsutherland) [inline pyretis/meta.yaml]: It looks like none of these are actually needed for building the package, so you should be able to delete everything but `python` and `setuptools` from here.

File: pyretis/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  run:
+  - python
+  - setuptools
+  - colorama
+  - tqdm
+  - future
```
Comment (dougalsutherland) [inline pyretis/meta.yaml]: The [setup.py](https://gitlab.com/pyretis/pyretis/blob/5acb1fd23ab8ecaa11f48c3c2c97126991e01daa/setup.py#L74) seems to require `docutils`, which isn't included here, and not require `future`?

**Comment by jakirkham** [comment]
Please move to the `recipe` directory. Thanks.

File: pyretis/meta.yaml (hunk 1 above)
Comment (EnricoRiccardi) [inline pyretis/meta.yaml]: done

File: pyretis/meta.yaml (hunk 3 above)
Comment (EnricoRiccardi) [inline pyretis/meta.yaml]: done all suggestions! thanks :)

File: recipes/pyretis/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+build:
+  number: 0
+  skip: True  # [py<3]
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+  - python
+  - setuptools
```
Comment (jakirkham) [inline recipes/pyretis/meta.yaml]: Please indent these 2 more spaces.

File: recipes/pyretis/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  run:
+  - python
+  - setuptools
+  - colorama
+  - tqdm
+  - docutils
+  - numpy
+  - scipy
+  - matplotlib
+  - jinja2
```
Comment (jakirkham) [inline recipes/pyretis/meta.yaml]: Please indent these 2 more spaces.

File: recipes/pyretis/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+test:
+  imports:
+  - pyretis
```
Comment (jakirkham) [inline recipes/pyretis/meta.yaml]: Please indent these 2 more spaces.

File: recipes/pyretis/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+about:
+  home: http://www.pyretis.org
+  license: LGPL-2.1 or later
```
Comment (jakirkham) [inline recipes/pyretis/meta.yaml]: ` or later` is implied by `LGPL`. So please drop that.

File: recipes/pyretis/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,47 @@
+{% set name = "pyretis" %}
+{% set version = "0.9.2.dev2" %}
```
Comment (jakirkham) [inline recipes/pyretis/meta.yaml]: Is there a non-`dev` release that we can use (i.e. stable release).

File: recipes/pyretis/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+about:
+  home: http://www.pyretis.org
+  license: LGPL-2.1 or later
+  summary: 'PyRETIS: A well-done, medium-sized python library for rare events'
+
+extra:
+  recipe-maintainers:
+  - EnricoRiccardi
+  - andersle
```
Comment (jakirkham) [inline recipes/pyretis/meta.yaml]: Please indent these 2 more spaces.

File: recipes/pyretis/meta.yaml (hunk 8 above)
Comment (EnricoRiccardi) [inline recipes/pyretis/meta.yaml]: The version 1.00 will be released soon on pypt. I tried to link the gitlab directly but I didn't manage to make it lint... If you have time to assist me I can try to make a new pull request for that, but I am afraid it is getting dispersive.

Comment (EnricoRiccardi) [inline recipes/pyretis/meta.yaml]: (in the while, I thank you for the help given already!)

Comment (EnricoRiccardi) [inline recipes/pyretis/meta.yaml]: after I did the previous suggestions indicated, the tests  
ci/circleci 
continuous-integration/appveyor/pr
continuous-integration/travis-ci/pr

failed

Comment (jakirkham) [inline recipes/pyretis/meta.yaml]: If there is not a non-`dev` release, that's fine. We try to distinguish `dev` releases (by releasing under a different label) if possible to avoid disruption for end users. This only really matters if there are non-`dev` releases that we can package. However it sounds like there are only `dev` releases (this is confirmed by PyPI). So there is nothing more to do here.

TL;DR Don't worry about it. This is fine.

**Comment by jakirkham** [comment]
The error seems to be something specific to this code. `FileNotFoundError` is Python 3 only IIRC.

```
File "setup.py", line 45, in <module>
    except FileNotFoundError:
NameError: name 'FileNotFoundError' is not defined
```

File: recipes/pyretis/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [py<3]
```
Comment (jakirkham) [inline recipes/pyretis/meta.yaml]: `py<3` -> `py2k`.

Comment (EnricoRiccardi) [inline recipes/pyretis/meta.yaml]: done and tests succesfull. Thanks!

**Comment by EnricoRiccardi** [comment]
what should be the next step to merge_?

**Comment by EnricoRiccardi** [comment]
what should be the next step to merge_?

File: recipes/pyretis/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+  skip: True  # [py2k]
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+  run:
+    - python
+    - setuptools
```
Comment (mwcraig) [inline recipes/pyretis/meta.yaml]: Please remove setuptools from the run dependencies.

**Comment by mwcraig** [comment]
@EnricoRiccardi -- sorry for the delay. One more minor change. Please ping in ~1 or 2 days if this hasn't merged by then. 

@ andersle -- could you please confirm that you are willing ot be a maintainer on this recipe?

File: recipes/pyretis/meta.yaml (hunk 11 above)
Comment (EnricoRiccardi) [inline recipes/pyretis/meta.yaml]: done

**Comment by EnricoRiccardi** [comment]
@andersle

**Comment by andersle** [comment]
@mwcraig: Yes, I can be a maintainer for the recipe.

**Comment by dougalsutherland** [comment]
@EnricoRiccardi Looks like https://github.com/conda-forge/staged-recipes/pull/3380/commits/e45434a11591edb980ae3e23d1a89c38bba3e8ce removed setuptools from the build dependencies and left it in the run dependencies; it should be the other way around. (This caused the CircleCI failure.)

**Comment by EnricoRiccardi** [comment]
fixed! Thanks

**Comment by EnricoRiccardi** [comment]
thank to you for the help and patience and support.

**Comment by mwcraig** [comment]
No problem, thanks for adding the recipe!

File: recipes/pyretis/meta.yaml (hunk 8 above)
Comment (EnricoRiccardi) [reply recipes/pyretis/meta.yaml]: Hi,

So finally we have the 1.0.0 not dev version. How to make an upgrade? I tried to fork and update, it has then been merged, but I think only to my branch?

Comment (dougalsutherland) [reply recipes/pyretis/meta.yaml]: You should send a pull request against https://github.com/conda-forge/pyretis-feedstock.
